src/pages/index.js

```diff
@@ -32,23 +42,120 @@ addPostButton.addEventListener("click", () => {
   postFormPopup.open();
 });
 
+//Handle Submit functions
+function handleSubmit(request, popupModal, popupForm, renderingText, postText) {
+  renderLoading(true, popupForm, renderingText, postText);
+  request()
+    .then(() => {
+      popupModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      renderLoading(false, popupForm, renderingText, postText);
+    });
+}
+
+const handleProfilePictureSubmit = ({ picture }) => {
+  function makeRequest() {
+    return api.updateAvatar(picture).then((profileInfo) => {
+      profilePicture.src = profileInfo.avatar;
+    });
+  }
+
+  handleSubmit(makeRequest, profilePictureFormPopup, editProfilePictureForm);
+};
+
 const handleProfileFormSubmit = ({ name, job }) => {
-  userInfo.setUserInfo({ name, job });
-  profileFormPopup.close();
+  function makeRequest() {
+    return api.updateProfileInfo({ name, about: job }).then((newData) => {
+      userInfo.setUserInfo({
+        name: newData.name,
+        job: newData.about,
+      });
+    });
+  }
+
+  handleSubmit(makeRequest, profileFormPopup, editProfileForm);
 };
 
 const handlePostFormSubmit = ({ title, url }) => {
   const newPost = {
     name: title,
     link: url,
   };
-  const newCard = createCard(newPost);
-  cardSection.addNewItem(newCard);
-  postFormPopup.close();
+
+  function makeRequest() {
+    return api.createApiCard(newPost).then((createdCard) => {
+      cardSection.addNewItem(createCard(createdCard));
+    });
+  }
+
+  handleSubmit(makeRequest, postFormPopup, addPostForm);
 
   formValidators[selectors.addPostForm].disableButton();
 };
 
+const handleDelete = (card) => {
+  confirmDeletePopup.open();
+  formValidators[selectors.confirmDeleteForm].enableButton();
+  confirmDeletePopup.setSubmitFunction(() => {
+    function makeRequest() {
+      return api.deleteCard(card._data._id).then(() => {
+        card.handleDeletePost();
+      });
+    }
+
+    handleSubmit(
+      makeRequest,
+      confirmDeletePopup,
+      confirmDeleteForm,
+      "Deleting...",
+      "Delete"
+    );
+  });
+};
+
+const handleLike = (card) => {
+  const updateLikeState = () => {
+    card._data.isLiked = !card._data.isLiked;
```

Comment:
`_data`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.